models/progresses.js

```diff
@@ -0,0 +1,138 @@
+const { Conflict } = require("http-errors");
+const fireStore = require("../utils/firestore");
+const progressesCollection = fireStore.collection("progresses");
+const { fetchTask } = require("./tasks");
+const { fetchUser } = require("./users");
+
+/**
+ * Adds a new progress document for the given user or task, with a limit of one progress document per day.
+ * @param progressData {object} The data to be added. It should be an object containing key-value pairs of the fields to be added, including a "type" field set to either "user" or "task".
+ * @returns {Promise<object>} A Promise that resolves with the added progress document object, or rejects with an error object if the add operation fails.
+ * @throws {Error} If a progress document has already been created for the given user or task on the current day.
+ **/
+const createProgressDocument = async (progressData) => {
+  const { type, userId, taskId } = progressData;
+  const createdAtTimestamp = new Date().getTime();
+  const currentHourIST = new Date().getUTCHours() + 5.5; // IST offset is UTC+5:30
+  const isBefore6amIST = currentHourIST < 6;
```

Comment:
Are we documenting this at any place that we are doing calculations with respect to IST?

Reply:
I had a discussion with @ankushdharkar about this matter during the morning standup. Currently, he suggested adding this as a comment directly in the code. Please let me know if you have any other recommendations.

Reply:
Yes its a 24 hours format 
![image](https://github.com/Real-Dev-Squad/website-backend/assets/97341921/09f56c61-c0ca-471c-929f-f0998811f12b)

Reply:
> > I had a discussion with @ankushdharkar about this matter during the morning standup. Currently, he suggested adding this as a comment directly in the code. Please let me know if you have any other recommendations.
> 
> For now, we can just add a comment in the code wherever we are using this.

Yes I have done that hence resolving this comment.